### PR TITLE
WIP Question type for storage forecast

### PR DESCRIPTION
**Pull request by vbhavdal**

### CODE REVIEW

File: src/easydmp/dmpt/forms.py [hunk 1]
```diff
@@ -581,6 +582,44 @@ def serialize_subform(self):
 )
 
 
+class StorageForecastForm(forms.Form):
+    """
+    A row in the "matrix" for a 5 year storage forecast
+    """
+
+    def __init__(self, year=None, question=None, *args, **kwargs):
```
Comment (hmpf) [inline src/easydmp/dmpt/forms.py:590]: Methods after attributes, please.

**Comment by hmpf** [comment]
Whenever you add something to INPUT_TYPES, django insist on a noop migration. I *really* need to look into fixing that... Had some ideas over the weekend.

**Comment by hmpf** [comment]
Urgh, html.

The remaining problem is what browser X does given form attributes like `required` and `disabled`. The input with type "number" (storage estimate) does not work with `required`, and it seems `disabled` acts weirdly with `year`. Basically what happens is you cannot submit, there's no error messages, and you cannot save.

File: src/easydmp/dmpt/models.py [hunk 2]
```diff
@@ -59,6 +59,7 @@
     'multinamedurlonetext',
     'multidmptypedreasononetext',
     'multirdacostonetext',
+    'storageforecast'
```
Comment (hmpf) [inline src/easydmp/dmpt/models.py:62]: This should probably be "multistorageforecastonetext" which I suspect is longer than the input_type is allowed to be. Luckily, using a data migration to fix this later is easy.

.. I think the "onetext" should be a parameter on the field anyway. It is whether the answer is munged into a sentence in the html export. If there are more than three, a list would be better and we might also say a sentence is only used if there are less than three.

File: src/easydmp/dmpt/models.py [hunk 3]
```diff
@@ -2146,6 +2147,58 @@ def validate_choice(self, data):
...
+
+    def save(self, *args, **kwargs):
+        self.input_type = 'storageforecast'
+        super().save(*args, **kwargs)
+
+    def get_canned_answer(self, choice, **kwargs):
+        if not choice:
+            return self.get_optional_canned_answer()
+
+        framing_text = self.framing_text if self.framing_text else self.DEFAULT_FRAMING_TEXT
```
Comment (vbhavdal) [inline src/easydmp/dmpt/models.py:2179]: self.framing_text or self.DEFAULT_FRAMING_TEXT?

Comment (hmpf) [reply src/easydmp/dmpt/models.py:2179]: self.framing_text is an optional field on the model IIRC. `DEFAULT_FRAMING_TEXT` needs to be set to something sensible regardless.

Comment (vbhavdal) [reply src/easydmp/dmpt/models.py:2179]: I meant can't this be a little bit shorter as `framing_text = self.framing_text or self.DEFAULT_FRAMING_TEXT` ? :)

Comment (hmpf) [reply src/easydmp/dmpt/models.py:2177]: I've been trying to train myself to use the "offical" trinary operator, which is `x if y else z`, but, hm..

**Pull request merged**